libs/amp-instrumentation/src/amp_instrumentation/_bootstrap/initialization.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+#
+# WSO2 LLC. licenses this file to you under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Instrumentation initialization logic.
+This module contains the core initialization function for instrumentation.
+"""
+
+import os
+import logging
+import threading
+from . import environment_variables as env_vars
+
+logger = logging.getLogger(__name__)
+
+# Track initialization state with thread safety
+_initialized = False
+_init_lock = threading.Lock()
+
+
+class ConfigurationError(Exception):
+    """Raised when required configuration is missing or invalid."""
+    pass
+
+
+def _get_required_env_var(var_name: str) -> str:
+    """
+    Get a required environment variable or raise ConfigurationError.
+
+    Raises:
+        ConfigurationError: If the variable is missing or empty.
+    """
+    value = os.getenv(var_name)
+    if not value or not value.strip():
+        raise ConfigurationError(
+            f"Required environment variable '{var_name}' is not set or is empty. "
+            f"Please set this variable before running the application."
+        )
+    return value.strip()
```

Comment:
Instead generalising this it can be more useful if we can give specific instructions on how to get each env variable (e.g. API key is by registering agent at WSO2 AMP)

Reply:
no need to change this now. we can improve later